src/features/query-bar/__tests__/Condition-test.js

```diff
@@ -29,43 +29,69 @@ describe('features/query-bar/components/filter/Condition', () => {
         });
     });
 
-    describe('updateSelectedField()', () => {
+    describe('updateColumnField()', () => {
+        test('should select a column', () => {
+            const condition = initialCondition;
+            const columnId = '1';
+            const option = {
+                type: 'string',
+                value: columnId,
+            };
+            const onColumnChange = jest.fn();
+            const wrapper = getWrapper({
+                onColumnChange,
+            });
+
+            wrapper
+                .find('SingleSelectField')
+                .at(0)
+                .simulate('change', option);
+
+            expect(onColumnChange).toHaveBeenCalledWith(condition, columnId);
+        });
+    });
+
+    describe('updateOperatorField()', () => {
         const displayText = 'Vendor Name';
-        const index = 0;
         const condition = initialCondition;
         const value = 0;
         const option = {
             displayText,
-            type: 'string',
             value,
         };
-        const valueType = 'string';
 
-        test.each`
-            description                    | fieldType     | displayTextType          | keyType
-            ${'should select a column'}    | ${'column'}   | ${'columnDisplayText'}   | ${COLUMN_KEY}
-            ${'should select an operator'} | ${'operator'} | ${'operatorDisplayText'} | ${OPERATOR_KEY}
-            ${'should select a value'}     | ${'value'}    | ${'valueDisplayText'}    | ${VALUE_KEY}
-        `('$description', ({ fieldType, displayTextType, keyType }) => {
+        test('should select an operator', () => {
+            const onColumnChange = jest.fn();
             const onFieldChange = jest.fn();
             const wrapper = getWrapper({
                 onFieldChange,
+                onColumnChange,
             });
 
             wrapper
                 .find('SingleSelectField')
                 .at(1)
-                .simulate('change', option, fieldType);
-
-            expect(onFieldChange).toHaveBeenCalledWith(
-                index,
-                condition,
-                displayText,
-                displayTextType,
-                value,
-                keyType,
-                valueType,
-            );
+                .simulate('change', option);
+            expect(onFieldChange).toHaveBeenCalledWith(condition, value, OPERATOR);
+        });
+    });
+
+    describe('updateSelectedField()', () => {
+        const condition = initialCondition;
+        const displayText = 'Vendor Name';
+        const value = 0;
+        const option = {
+            displayText,
+            value,
+        };
+
+        test('should update value field', () => {
+            const onFieldChange = jest.fn();
+            const wrapper = getWrapper({ onFieldChange });
+
+            wrapper.find('ValueField').prop('updateSelectedField')(option);
```

Comment:
Shouldn't updateSelectedField be named onFieldChange or onValueChange?

Can you take go through everything named "update..." and check if it needs to be changed?

grep -ri update FOLDER